### ISSUE TITLE
Is ElasticSearch output package backwards compatible to Elastic v2.x?

### ISSUE THREAD

**Issue by ben-burton** [Azure/diagnostics-eventflow#211]
The docs mention:

>Note: Nuget package version 1.x supports Elasticsearch version 2.x. Nuget package version 2.x supports Elasticsearch version 5.x

Does this mean the latest packages have backwards compatibility for Elastic search v2.x or do I need to downgrade my NuGet package? I can't currently downgrade to an earlier 1.x package due to #185 - I'm using ASP.NET Core Stateless Service with Service Fabric.

I'm getting 
```
ElasticSearchOutput: request resulted in an error: Debug information: Invalid NEST response built from a unsuccessful low level call on POST: /_bulk
```
message show up as a warning in Service Fabric Explorer which I suspect may be due to a compatibility issue?

Thanks

**Comment by karolz-ms** [comment]
@ben-burton Hi!

The documentation note mentioned by you is directly related to ES version compatibility between NEST libraries EventFlow ES output uses. ES output 1.x uses NEST 2.x series; ES output 2.x uses NEST 5.x. My understanding is that NEST 5.x series of libraries does not support Elasticsearch 2.x, only Elasticsearch 5.x and above. Seems like the error you are getting is consistent with this, unfortunately.

[Issue 185](https://github.com/Azure/diagnostics-eventflow/issues/185) should not be blocking--manually referencing a newer version of NEST package (from the NEST 2.x series, if you intend to use Elasticsearch 2.x/EventFlow ES output 1.x) should eliminate the warning. 

Are you using .NET full framework or .NET Core to implement your Service Fabric service. If .NET Core, what version of .NET Core: 1.1. or 2.0?

**Comment by ben-burton** [comment]
Hi @karolz-ms, thanks for the prompt reply.

Thanks for confirming. I'm using .NET Core 2.0. I've just tried adding a 2.x NEST package from NuGet and then installing v1.0 of EventFlow.Outputs.ElasticSearch afterwards as you suggest. I'm now able to effectively downgrade this way so I can try with the lower version.

I tried this, running my Service Fabric cluster locally using the downgraded version, but I'm still seeing the same message as above. As you explain, given the version relates to the version of NEST. Looking at https://github.com/elastic/elasticsearch-net I'm wondering whether NEST supports my version of Elastic Search at all. As I've tried the lowest version EventFlow ElasticSearch output package, I think I'd have to assume my version of Elastic is the problem here.

Thanks for your help
